src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -433,10 +425,14 @@ bool emberAfIcdManagementClusterUnregisterClientCallback(CommandHandler * comman
 bool emberAfIcdManagementClusterStayActiveRequestCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
                                                           const Commands::StayActiveRequest::DecodableType & commandData)
 {
-    ICDManagementServer server;
-    InteractionModel::Status status = server.StayActiveRequest(commandObj->GetAccessingFabricIndex());
-
-    commandObj->AddStatus(commandPath, status);
+// Note: We only need this #if statement for platform examples that enable the ICD management server without building the sample
+// as an ICD. Since this is not spec compliant, we should remove this #if statement once we stop compiling the ICD management
+// server in those examples.
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
```

Comment:
Should we add another build flag for the StayActiveRequest to mimick the CIP and UAT defines?

Reply:
I don't think so. It makes the flow from enabling this command more complex because you need to enable it in zap, then to ensure the define is here with whatever build flag we add.
If we do not enable it in zap`emberAfIcdManagementClusterStayActiveRequestCallback` will be optimized out as unused and won't take space on the device.
The only reason to ifdef it at all is due to compilation errors when non-icd try to build this cluster.